app/controllers/jobseekers/job_applications_controller.rb

```diff
@@ -43,6 +43,27 @@ def destroy
                 success: t("messages.jobseekers.job_applications.draft_deleted", job_title: vacancy.job_title)
   end
 
+  def confirm_withdraw
+    raise ActionController::RoutingError, "Cannot withdraw non-submitted or non-shortlisted application" unless
+      job_application.status.in?(%w[shortlisted submitted])
+  end
+
+  def withdraw
+    raise ActionController::RoutingError, "Cannot withdraw non-submitted or non-shortlisted application" unless
+      job_application.status.in?(%w[shortlisted submitted])
+
+    if params[:commit] == t("buttons.cancel")
+      return redirect_to jobseekers_job_applications_path if URI(params[:origin]).path == jobseekers_job_applications_path
```

Comment:
there is a line of code after the return?

Reply:
there's an
```ruby
if
```
after it